src/history.rs

```diff
@@ -164,7 +185,13 @@ impl<'a> DynamicHistory<'a> {
             clocks,
             log,
             log_items_missed: 0,
+            restart_counter: RestartSequenceCounter::new(restart_counter),
         };
+        if history.restart_counter.is_tracking_restarts() {
+            let next_persistent_epoch = history.restart_counter.next_sequence_id(history.probe_id);
```

Comment:
These two lines are repeated almost semantically verbatim below in `increment_local_clock`. Consider making a function which captures this shared behavior.